brainwidemap/decoding/functions/process_outputs.py

```diff
@@ -0,0 +1,382 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Dec  2 08:58:41 2022
+
+@author: bensonb
+"""
+import numpy as np
+import pandas as pd
+from sklearn.metrics import r2_score, balanced_accuracy_score
+
+def gini(x, weights=None):
+    '''
+    Index that measures sparsity across a set of values, x.
+    If values in x are roughly constant, then this index is near 0.
+    If values in x have a large spread, then this index is near 1.
+
+    Implementation copied from 
+    https://stackoverflow.com/questions/48999542/more-efficient-weighted-gini-coefficient-in-python
+
+    Parameters
+    ----------
+    x : 1-d array of values
+        .
+    weights : 1-d array, floats
+        weightings corresponding to values of x. The default is None.
+
+    Returns
+    -------
+    float
+
+    '''
+    if weights is None:
+        weights = np.ones_like(x)
+    # Calculate mean absolute deviation in two steps, for weights.
+    count = np.multiply.outer(weights, weights)
+    mad = np.abs(np.subtract.outer(x, x) * count).sum() / count.sum()
+    rmad = mad / np.average(x, weights=weights)
+    # Gini equals half the relative mean absolute deviation.
+    return 0.5 * rmad
+
+def fix_pd_regions(res):
```

Comment:
Function can be removed (see 04_fromat_results.py)

Reply:
@juhuntenburg we decided to keep because this function takes a long time to execute and printing progress is useful